modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2259 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// Navier-Stokes includes
+#include "NSFVAction.h"
+#include "NS.h"
+
+// MOOSE includes
+#include "FEProblem.h"
+#include "MooseObject.h"
+#include "NonlinearSystemBase.h"
+#include "RelationshipManager.h"
+
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_variables");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_user_objects");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_ics");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_kernels");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_bcs");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_material");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_pps");
+registerMooseAction("NavierStokesApp", NSFVAction, "add_navier_stokes_materials");
+
+InputParameters
+NSFVAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("This class allows us to set up Navier-Stokes equations for porous "
+                             "medium or clean fluid flows using incompressible or weakly "
+                             "compressible approximations with a finite volume discretization.");
+
+  /**
+   * General parameters used to set up the simulation.
+   */
+  params.addParam<std::vector<SubdomainName>>(
+      "block", "The list of blocks on which NS equations are defined on");
+
+  MooseEnum sim_type("steady-state transient", "steady-state");
```

Comment:
I don't have a strong preference on this. I'm sure real users will use more transient than steady-state. I myself do more steady but that is because I'm a developer, not a real user.

Reply:
funny that your comment shows up before mine.

Yes querying the executioner would definitely be best

Reply:
Added! Now, we test `_problem->isTransient()`.